src/server/game/Spells/Auras/SpellAuras.cpp

```diff
@@ -2304,14 +2308,14 @@ void UnitAura::FillTargetMap(std::unordered_map<Unit*, uint32>& targets, Unit* c
                         selectionType = TARGET_CHECK_ENEMY;
                         break;
                     case SPELL_EFFECT_APPLY_AREA_AURA_PET:
-                        if (!condList || sConditionMgr->IsObjectMeetToConditions(GetUnitOwner(), GetUnitOwner(), *condList))
+                        if (!condList || sConditionMgr->IsObjectMeetToConditions(GetUnitOwner(), caster, *condList))
                             units.push_back(GetUnitOwner());
                         /* fallthrough */
                     case SPELL_EFFECT_APPLY_AREA_AURA_OWNER:
                     {
                         if (Unit* owner = GetUnitOwner()->GetCharmerOrOwner())
                             if (GetUnitOwner()->IsWithinDistInMap(owner, radius))
-                                if (!condList || sConditionMgr->IsObjectMeetToConditions(owner, GetUnitOwner(), *condList))
+                                if (!condList || sConditionMgr->IsObjectMeetToConditions(owner, caster, *condList))
```

Comment:
The same change must be appled to the two cases not present in 3.3.5 branch: SPELL_EFFECT_APPLY_AURA_ON_PET and SPELL_EFFECT_APPLY_AREA_AURA_SUMMONS

Reply:
is it now correct?